frontends/web/src/hooks/matchmedia.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2022 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useState } from 'react';
+
+export const useMatchMedia = (query: string): boolean => {
+  const getMatches = (query: string): boolean => {
+    if (typeof window !== 'undefined') {
+      return window.matchMedia(query).matches;
+    }
+    return false;
+  };
```

Comment:
Can getMatches be moved out of the hook before useMatchMedia?
and `window` should always be defined, or not?

```diff
diff --git a/frontends/web/src/hooks/matchmedia.ts b/frontends/web/src/hooks/matchmedia.ts
index 9aca01c9c..07a38d03e 100644
--- a/frontends/web/src/hooks/matchmedia.ts
+++ b/frontends/web/src/hooks/matchmedia.ts
@@ -16,14 +16,9 @@
 
 import { useEffect, useState } from 'react';
 
-export const useMatchMedia = (query: string): boolean => {
-  const getMatches = (query: string): boolean => {
-    if (typeof window !== 'undefined') {
-      return window.matchMedia(query).matches;
-    }
-    return false;
-  };
+const getMatches = (query: string): boolean => matchMedia(query).matches;
 
+export const useMatchMedia = (query: string): boolean => {
   const [matches, setMatches] = useState<boolean>(getMatches(query));
```

Reply:
Yes it can, and... yeah it **should** always be :) I however sometimes get window `undefined` but it's usually for web projects that requires SSR. Shouldn't be the case here..? Anyhow I'll remove it, thx!

Reply:
no SSR due to no server :)